src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final String MESSAGE = "There is not such element.";
+    private static final String MESSAGE_OUT = "Out of bounds ";
+    private static final double NUMBER = 1.5;
+    private int size;
+    private Object[] elements;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            grow();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index, size + 1);
+        if (size == elements.length) {
+            grow();
+        }
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, size);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index, size);
+        elements[index] = value;
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, size);
+        T removedObject = (T) elements[index];
+        System.arraycopy(elements, index + 1, elements, index, size - index - 1);
+        size--;
+        return removedObject;
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public T remove(T element) {
-        return null;
+        Object removedObject;
+        for (int i = 0; i < size; i++) {
+            if (element == elements[i] || elements[i] != null && elements[i].equals(element)) {
+                removedObject = elements[i];
+                remove(i);
+                return (T) removedObject;
```

Comment:
don't forget to remove `removedObject` variable
```suggestion
                return remove(i);
```